packages/react-dom/src/__tests__/ReactDOMInput-test.js

```diff
@@ -794,13 +815,16 @@ describe('ReactDOMInput', () => {
       <input type="text" value="" onChange={emptyFunction} />,
       container,
     );
+    const node = container.firstChild;
+    expect(isValueDirty(node)).toBe(false);
+
     ReactDOM.render(
       <input type="text" value={0} onChange={emptyFunction} />,
       container,
     );
 
-    const node = container.firstChild;
     expect(node.value).toBe('0');
+    expect(isValueDirty(node)).toBe(true);
```

Comment:
Interestingly, this sole test fails on main with disableInputAttributeSyncing off:

```
  ● ReactDOMInput › should properly transition from an empty value to 0

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false

      825 |
      826 |     expect(node.value).toBe('0');
    > 827 |     expect(isValueDirty(node)).toBe(true);
          |                                ^
      828 |
      829 |     if (disableInputAttributeSyncing) {
      830 |       expect(node.hasAttribute('value')).toBe(false);
```

(It passes with the flag on.) So it regressed recently. Possibly related to the bug @rickhanlonii was debugging.

Reply:
expect head